manual/embedded-examples/src/docs/dev/procedures.asciidoc

```diff
@@ -0,0 +1,495 @@
+| ListLabels            | CALL sys.db.labels()            | List all labels in the database.
+| ListRelationshipTypes | CALL sys.db.relationshipTypes() | List all relationship types in the database.
+| ListPropertyKeys      | CALL sys.db.propertyKeys()      | List all property keys in the database.
+|===
+
+
+== Developing Compiled Procedures
+
+[TIP]
+--
+The example discussed below is available as {procedure-template-url}[a repository on Github].
+To get started quickly you can fork the repository and work with the code as you follow along in the guide below.
+--
+
+Compiled procedures are written in the Java programming language.
+While you can develop Compiled Procedures using a regular text editor it is recommended to use a dedicated Java development environment, like IntelliJ IDEA or Eclipse.
+
+To develop a new Procedure, you need to produce a `jar` file that contains the procedure along with any non-Neo4j dependencies it has.
+This file is then deployed in the `plugin` directory of each database in your cluster.
+The procedure will become available in each instance after restart.
+
+The general process for building a new Compiled Procedure is:
+
+* Set up a new Java project in your IDE
+* Write a test for the Procedure using Neo4j Harness
+* Write a procedure that makes your test pass
+* Build a `jar`-file from your project and deploy it
+
+
+=== Setting up a new project
+
+You can set up your project in any way that allows you to produce a Jar file.
+Below is an example of how to configure the Maven build system to compile a Procedure and produce a `jar` file.
+The Maven _.pom_-file is broken into pieces with comments for readability, the whole file is available from the {procedure-template-url}[Neo4j Procedure Template] repository.
+
+.Setting up a project with Maven
+[source, xml]
+----
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
+                     http://maven.apache.org/xsd/maven-4.0.0.xsd">
+ <modelVersion>4.0.0</modelVersion>
+
+ <groupId>org.neo4j.example</groupId>
+ <artifactId>procedure-template</artifactId>
+ <version>1.0.0-SNAPSHOT</version>
+
+ <packaging>jar</packaging>
+ <name>Neo4j Procedure Template</name>
+ <description>A template project for building a Neo4j Procedure</description>
+
+ <properties>
+   <neo4j.version>3.0.0-SNAPSHOT</neo4j.version>
+ </properties>
+----
+
+Next the dependencies of our build are defined.
+The following two sections are included in the _pom.xml_ between opening and closing `<dependencies></dependencies>` tags.
+(See the linked template repository for the complete example.)
+
+The first dependency section includes the Procedure API that the procedure uses at runtime.
+The scope is set to `provided`, because at deployment in a Neo4j Instance, the API will be provided by Neo4j.
```

Comment:
_because **at** deployment_ ... I need to read that thrice to understand it. Do you mean _because when deployed in a Neo4j instance_?